csi/node_server.go

```diff
@@ -386,6 +386,28 @@ func (ns *NodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVol
 		return nil, err
 	}
 
+	// check if we need to resize the fs
+	// this is important since cloned volumes of bigger size don't trigger NodeExpandVolume
+	// therefore NodeExpandVolume is kind of redundant since we have to do this anyway
+	// some refs below for more details
+	// https://github.com/kubernetes/kubernetes/issues/94929
+	// https://github.com/kubernetes-sigs/aws-ebs-csi-driver/pull/753
+	resizer := mount.NewResizeFs(utilexec.New())
+	if needsResize, err := resizer.NeedResize(devicePath, targetPath); err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	} else if needsResize {
```

Comment:
NIT:
```suggestion
	needsResize, err := resizer.NeedResize(devicePath, targetPath); 
	if err != nil {
		return nil, status.Error(codes.Internal, err.Error())
	} 
	if needsResize {
```

Similar below

Reply:
I find the existing one more readable, it also reduces the scope of the variables.
But both approaches are fine, I tend to prefer the compacter code when the inner scope only has a couple of lines.

Reply:
OK. The suggestions I provide reduces the nesting. But in general both are fine. Let me retain the current implementation.